hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/client/ReadManyRequest.java

```diff
@@ -64,11 +64,13 @@ public int getClassId() {
     @Override
     protected Object filter(Object response) {
         ReadResultSetImpl readResultSet = (ReadResultSetImpl) response;
+
         int readCount = readResultSet.readCount();
         List<Data> items = new ArrayList<Data>(readCount);
-        for (Data item : readResultSet.getDataItems()) {
-            items.add(item);
+        for (int k = 0; k < readResultSet.size(); k++) {
+            items.add(readResultSet.getDataItems()[k]);
```

Comment:
perhaps C2 will optimize this away, but I'd still remove the `getDataItems()` outside of the loop.